### ISSUE TITLE
Add commit SHA to the nightly builds

### ISSUE THREAD

**Issue by Utumno** [wrye-bash/wrye-bash#675]
Doesn't need to be the full sha ten chars should be enough - helpful to track where a nightly was built from since we (wisely) stopped pushing the build commit to nightly.

**Comment by Infernio** [comment]
Definitely a good idea. Note the comment though:

https://github.com/wrye-bash/wrye-bash/blob/8a11027e57b942d5ae084a3a81bafdf4bd5884b6/Mopy/bash/bass.py#L40

Some scripts rely on this for sure, not sure what else does. It might be best to create a new `app_version_extra` string constant and format with that, like this:

```python
f'{bass.AppVersion}-{bass.app_version_extra}'
```

Then log that in the BashBugDump instead of bare AppVersion, same with the title bar. Would also let us easily add `RC1` etc. to release candidates.

Maybe do the formatting right in bass? Not sure what's better.

**Comment by Metallicow** [comment]
... OK? Do I want this?
Is there a Dynamic way to update the 'built' status to a customer besides saying OK go here click that...

... other than noob-automated principles besides "Nightly" ...

**Comment by Utumno** [comment]
While working on this issue, I ended up digging deeper into the build pipeline: the current packaging approach relies heavily on recursive glob patterns, which makes the build artifact sensitive to whatever happens to exist in a developer’s working tree (very common in long-lived clones). In practice, the “fix” is often “do a fresh clone”, which becomes an extra manual step - exactly what the build process shouldn’t require. To make the build deterministic, I switched packaging to derive the file list from Git (pack exactly what we intend to ship). That naturally led to the next, harder step: replacing the manually maintained `RemoveFiles` NSIS macro with a programmatic variant derived from Git history, plus a small manual layer for dev/nightly-only byproducts and binaries. Since I’ve already implemented most of this, I won’t open a new issue; what remains is finishing the “delete historical leftovers” logic. The hardest part is the analysis, and I wrote a small Python [script](https://github.com/user-attachments/files/24994699/diff_del_files.py) to quantify current macro coverage and identify what’s uncovered/unused - the script compares the existing `RemoveFiles` macro lines against:

* `tracked(depth=1)` = paths tracked by current `HEAD`
* `tracked(all)` = paths ever tracked in history reachable from `HEAD`
* `untracked = tracked(all) - tracked(depth=1)` = “historical leftovers” we might want to delete from old installs

## TL;DR

For `Mopy/`, the macro already covers almost everything (2235/2251 historical leftovers). A programmatic version can (a) add coverage for the small uncovered set, and (b) keep (after minor edits) a subset of “unused” rules that still make sense as cleanup for dev/build byproducts.

<details>
<summary>Script output</summary>

```text
== Macro rule counts ==
delete_rules_total: 762 (exact=693, star=69)
delete_rules_mopy:  exact=682, star=68
delete_rules_other: exact=11, star=1
rmdir_rules_total:  31
rmdir_rules_mopy:   31
rmdir_rules_other:  0
== Repo tracked/untracked ==
tracked(depth=1):   644
tracked(all):       3144
untracked(all):     2523
untracked(Mopy):    2251
untracked(outside): 272
== Mopy coverage ==
need_to_cover: 2251
covered_rmdir:  1200
covered_exact:  639
covered_star:   396
uncovered:      16
uncovered_sample (up to 20):
  Mopy/Bash Patches/Fallout 3/taglist.txt
  Mopy/Bash Patches/Fallout New Vegas/taglist.txt
  Mopy/bash/_i18n/README.md
  Mopy/bash/compiled/libbsa32.dll
  Mopy/bash/compiled/libbsa64.dll
  Mopy/bash/compiled/loadorder32.dll
  Mopy/bash/images/people/cptmavi32.jpg
  Mopy/bash/images/people/utmnavi32.jpg
  Mopy/bash/images/people/vldaavi32.jpg
  Mopy/bash/images/plugin_checker.svg
  Mopy/bash/images/readme/mouse_gesture_animation.gif
  Mopy/bash/images/wrye_monkey_32.jpg
  Mopy/bash/images/wrye_monkey_64.jpg
  Mopy/bash/images/wrye_monkey_87.jpg
  Mopy/bash/images/wryemonkey16.jpg
  Mopy/bash_default.ini
== Outside-Mopy coverage ==
need_to_cover: 272
covered_rmdir:  0
covered_exact:  10
covered_star:   1
uncovered:      261
uncovered_sample (up to 20):
  .coveragerc
  .gitattributes
  .github/ISSUE_TEMPLATE/config.yml
  .github/issue_template.md
  .github/pull_request_template.md
  .github/workflows/pr.yaml
  .github/workflows/update_pot.yaml
  .github/workflows/wb_ci.yaml
  .gitignore
  .gitmodules
  .pre-commit-config.yaml
  .project
  .pydevproject
  .weblate
  Contributing.md
  Data/ArchiveInvalidationInvalidated!.bsa
  Data/Bash Patches/Assorted to Cobl.csv
  Data/Bash Patches/Assorted_Exhaust.csv
  Data/Bash Patches/Bash_Groups.csv
  Data/Bash Patches/Bash_MFact.csv
== Macro rules with zero matches against `untracked` ==
unused_rmdir: 5 (mopy=5, outside=0)
unused_exact: 43 (mopy=42, outside=1)
unused_star:  3 (mopy=3, outside=0)
unused_rmdir_sample (up to 20):
  Mopy/Bash Patches/Enderal
  Mopy/bash/__pycache__
  Mopy/redist
  Mopy/taglists/Fallout4VR
  Mopy/taglists/SkyrimVR
unused_exact_sample (up to 100):
  Mopy/bash/balto
  Mopy/bash/bapio
  Mopy/bash/barbo
  Mopy/bash/bargo
  Mopy/bash/bashero
  Mopy/bash/basho
  Mopy/bash/basso
  Mopy/bash/belto
  Mopy/bash/bolto
  Mopy/bash/bosho
  Mopy/bash/breco
  Mopy/bash/busho
  Mopy/bash/bwebo
  Mopy/bash/cinto
  Mopy/bash/images/krita16.png
  Mopy/bash/images/krita24.png
  Mopy/bash/images/krita32.png
  Mopy/bash/images/readme/installers-wizard-1.jpg
  Mopy/bash/images/readme/installers-wizard-2.jpg
  Mopy/bash/images/readme/mods-feat-add-tags.png
  Mopy/bash/images/readme/mods-feat-change-mtime.png
  Mopy/bash/images/readme/mods-feat-del-tags.png
  Mopy/bash/images/readme/saves-2-rclick-save-5.png
  Mopy/bash/images/readme/saves-2-rclick-save-6.png
  Mopy/bash/images/readme/saves-2-rclick-save-7.png
  Mopy/bash/keywordWIZBAIN2o
  Mopy/bash/keywordWIZBAINo
  Mopy/bash/l10n/Chinese (Simplified).mo
  Mopy/bash/l10n/Chinese (Traditional).mo
  Mopy/bash/l10n/Italian.mo
  Mopy/bash/l10n/Japanese.mo
  Mopy/bash/l10n/Russian.mo
  Mopy/bash/l10n/de.mo
  Mopy/bash/l10n/pt_opt.mo
  Mopy/bash/l10n/sv.mo
  Mopy/bash/l10n/tr.mo
  Mopy/bash/libbsao
  Mopy/bash/settingsModuleo
  Mopy/bash/windowso
  Mopy/bash/wizSTCo
  Mopy/uninstall.exe
  Mopy/w9xpopen.exe
  scripts/built_taglist
unused_star_sample (up to 20):
  Mopy/bash/images/wrye_monkey_*.png
  Mopy/loot.*
  Mopy/loot_api.*
Outside-Mopy top roots (untracked):
  Data: 197
  scripts: 48
  .github: 6
  New Skyrim Forum thread starter.txt: 1
  pyproject.toml: 1
  Readme.md: 1
  Bug List thread Starter.txt: 1
  .gitattributes: 1
  .coveragerc: 1
  .weblate: 1
Outside-Mopy top roots (uncovered):
  Data: 197
  scripts: 43
  .github: 6
  pyproject.toml: 1
  .gitattributes: 1
  .weblate: 1
  .gitmodules: 1
  .project: 1
  requirements-scripts.txt: 1
  Readme.md: 1
```

</details>

## Uncovered `Mopy/` (16)

* Bash Patches taglists (folder naming mismatch vs current `RMDir`)

  * `Mopy/Bash Patches/Fallout 3/taglist.txt`
  * `Mopy/Bash Patches/Fallout New Vegas/taglist.txt`
* Compiled DLL leftovers

  * `Mopy/bash/compiled/{libbsa32.dll,libbsa64.dll,loadorder32.dll}`
* Docs/config

  * `Mopy/bash/_i18n/README.md`
  * `Mopy/bash_default.ini`
* Other assets

  * `Mopy/bash/images/plugin_checker.svg`
  * `Mopy/bash/images/readme/mouse_gesture_animation.gif`
* Monkey/people images

  * `Mopy/bash/images/wrye_monkey_{32,64,87}.jpg`
  * `Mopy/bash/images/wryemonkey16.jpg`
  * `Mopy/bash/images/people/{cptmavi32,utmnavi32,vldaavi32}.jpg`

## `unused_star_mopy`

* `Mopy/bash/images/wrye_monkey_*.png` doesn’t match the uncovered monkey files (`.jpg`, plus `wryemonkey16.jpg`).
* `Mopy/loot.*` and `Mopy/loot_api.*` look worth keeping.

## `unused_exact_mopy`

<details>
<summary>Grouped unused exact rules (Python list)</summary>

```python
unused_exact_keep = [ # Mopy

[diff_del_files.py](https://github.com/user-attachments/files/24994692/diff_del_files.py)

    # Old old files to delete (from before 294, the directory restructure)
    'Mopy/uninstall.exe',
    # Some files from an older version of the Standalone that made non-standard
    # compiled python file names (when loading python files present)
    'Mopy/bash/balto','Mopy/bash/bapio','Mopy/bash/barbo','Mopy/bash/bargo','Mopy/bash/bashero','Mopy/bash/basho','Mopy/bash/basso','Mopy/bash/belto','Mopy/bash/bolto','Mopy/bash/bosho','Mopy/bash/breco','Mopy/bash/busho','Mopy/bash/bwebo','Mopy/bash/cinto','Mopy/bash/libbsao','Mopy/bash/windowso',
    # image files were moved to Mopy\bash\images\tools as of version 300
    'Mopy/bash/images/krita16.png','Mopy/bash/images/krita24.png','Mopy/bash/images/krita32.png',
    # As of 307, the following are obsolete:
    'Mopy/bash/images/readme/installers-wizard-1.jpg','Mopy/bash/images/readme/installers-wizard-2.jpg','Mopy/bash/images/readme/mods-feat-add-tags.png','Mopy/bash/images/readme/mods-feat-change-mtime.png','Mopy/bash/images/readme/mods-feat-del-tags.png','Mopy/bash/images/readme/saves-2-rclick-save-5.png','Mopy/bash/images/readme/saves-2-rclick-save-6.png','Mopy/bash/images/readme/saves-2-rclick-save-7.png',
    # As of 301 the following are obsolete:
    'Mopy/bash/keywordWIZBAIN2o','Mopy/bash/keywordWIZBAINo','Mopy/bash/settingsModuleo','Mopy/bash/wizSTCo',
    # As of 308, translations use the .po extension and new names
    'Mopy/bash/l10n/Chinese (Simplified).mo','Mopy/bash/l10n/Chinese (Traditional).mo','Mopy/bash/l10n/Italian.mo','Mopy/bash/l10n/Japanese.mo','Mopy/bash/l10n/Russian.mo','Mopy/bash/l10n/de.mo','Mopy/bash/l10n/pt_opt.mo',
    # The .po's for these were only temporarily on dev, then got renamed
    'Mopy/bash/l10n/sv.mo','Mopy/bash/l10n/tr.mo',
    # As of 305, the following are obsolete:
    'Mopy/w9xpopen.exe',
]
```

</details>

## `unused_rmdir_mopy`

* `Mopy/Bash Patches/Enderal`
* `Mopy/bash/__pycache__`
* `Mopy/redist`
* `Mopy/taglists/Fallout4VR`
* `Mopy/taglists/SkyrimVR`

Edit: forgot to mention that as the analysis shows very few files were removed outside mopy - I will drop this entirely, only delete files inside mopy - the ones that were treated were mostly very old anyways.

**Comment by Utumno** [comment]
I moved the remaining build-related items to the bugburner (and tagged a few with [A-build](https://github.com/wrye-bash/wrye-bash/issues?q=label%3AA-build)). With the recent work making packaging deterministic, the build pipeline mostly no longer needs manual edits or keeping ad-hoc structures in sync, so this isn't an immediate priority.

If/when we tackle more build improvements, a nice follow-up would be running the installer build on GitHub Actions as well (nightlies are still assembled manually today). Longer-term, moving repetitive build/NSIS scripting into Python seems like a good way to keep refactoring the build.
